tests/test_csc.py

```diff
@@ -0,0 +1,302 @@
+        """
+        self.patch.return_value.AVS_Activate.return_value = AvsReturnCode.invalidHandle.value
+        async with Harness(initial_state=salobj.State.STANDBY) as harness:
+            # Check that we are properly in STANDBY at the start
+            await self.check_summaryState(harness.remote, salobj.State.STANDBY)
+
+            msg = "Failed to connect"
+            with salobj.assertRaisesAckError(ack=salobj.SalRetCode.CMD_FAILED, result_contains=msg):
+                await harness.remote.cmd_start.start(timeout=STD_TIMEOUT)
+            state = await harness.remote.evt_summaryState.next(flush=False, timeout=STD_TIMEOUT)
+            self.assertEqual(state.summaryState, salobj.State.FAULT)
+            error = await harness.remote.evt_errorCode.next(flush=False, timeout=STD_TIMEOUT)
+            self.assertIn("RuntimeError('Invalid device handle; cannot activate device", error.errorReport)
+            self.assertIsNone(harness.csc.device)
+
+    async def test_expose(self):
+        """Test that we can take an exposure and that appropriate events are
+        emitted.
+        """
+        async with Harness(initial_state=salobj.State.ENABLED) as harness:
+            # Check that we are properly in ENABLED at the start
+            await self.check_summaryState(harness.remote, salobj.State.ENABLED)
+
+            duration = 2  # seconds
+            task = asyncio.create_task(harness.remote.cmd_expose.set_start(timeout=STD_TIMEOUT+duration,
+                                                                           duration=duration))
+            await asyncio.sleep(0.1)  # give the event loop time to start
+            # the exposure state should be Integrating during the exposure
+            await self.check_exposureState(harness.remote, ExposureState.INTEGRATING)
+            await task
+            # the exposure state should be Done after a successful exposure
+            await self.check_exposureState(harness.remote, ExposureState.DONE)
+
+    async def test_expose_fails(self):
+        """Test that a failed exposure puts us in the FAULT state, which will
+        disconnect the device.
+        """
+        self.patch.return_value.AVS_GetScopeData.side_effect = None
+        self.patch.return_value.AVS_GetScopeData.return_value = AvsReturnCode.ERR_INVALID_MEAS_DATA.value
+        async with Harness(initial_state=salobj.State.ENABLED) as harness:
+            # Check that we are properly in ENABLED at the start
+            await self.check_summaryState(harness.remote, salobj.State.ENABLED)
+
+            msg = "Failed to take exposure"
+            with salobj.assertRaisesAckError(ack=salobj.SalRetCode.CMD_FAILED, result_contains=msg):
+                await harness.remote.cmd_expose.set_start(timeout=STD_TIMEOUT, duration=.5)
+            # the exposure state should be Integrating during the exposure
+            await self.check_exposureState(harness.remote, ExposureState.INTEGRATING)
```

Comment:
I would appreciate a bit more detailed explanation (in code comments) of the failure you are producing -- enough that it becomes obvious why INTEGRATING will be output before FAILED.